### PR TITLE
feat(transparent-proxy): allow configure transparent proxy from config file

### PR DESCRIPTION
**Pull request by bartsmykla**
This PR implements MADR: [Streamlining Transparent Proxy Configuration (Phase 1): Enable Configuration via Config Files and Environment Variables](https://docs.google.com/document/d/1auMWZDEEbuu_kBfrpP5VfxbxJqN9FTxXysZRLz4g4Wg)

> [!NOTE]  
> Since this PR includes several changes that were necessary to introduce the described functionality, I recommend reviewing it commit by commit

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues
  - https://github.com/kumahq/kuma/issues/8324
  - [MADR 061: Streamlining Transparent Proxy Configuration (Phase 1): Enable Configuration via Config Files and Environment Variables](https://docs.google.com/document/d/1auMWZDEEbuu_kBfrpP5VfxbxJqN9FTxXysZRLz4g4Wg)
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS
  - It won't
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s)
  - Tests updated as well as manually tested
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)?
  - There is no need
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label)
  - There is no need

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/config/loader.go [hunk 1]
```diff
@@ -1,56 +1,141 @@
...
+func (l *Loader) Load(stdin io.Reader) error {
+	switch l.Filename {
+	case "-":
+		return l.LoadReader(stdin)
+	case "":
+		return l.LoadBytes([]byte(l.Content))
+	default:
+		return l.LoadFile(l.Filename)
 	}
+}
```
Comment (Icarus9913) [inline pkg/config/loader.go:60]: Just curious of when we would call this method? Also we expose `Filename` and `Content` properties as Public but we don't have these two parameters in the NewLoader functions.

So, is this a way to allow us to use nonstandard `Loader`? Could you please add some comments for these case conditions meaning?

File: pkg/transparentproxy/config/config.go [hunk 2]
```diff
@@ -57,6 +72,27 @@ func (c *Owner) Set(s string) error {
...
+	var jsonValue interface{}
+
+	if err := json.Unmarshal(bs, &jsonValue); err != nil {
+		return err
+	}
+
+	switch typedValue := jsonValue.(type) {
+	case string, float64:
+		return c.Set(fmt.Sprint(jsonValue))
+	case map[string]interface{}:
```
Comment (jijiechen) [inline pkg/transparentproxy/config/config.go:85]: This looks like a magic. Why do we need to handle map and key "uid" specially? 
May be a comment is helpful if it's not a big deal.

File: pkg/transparentproxy/config/config.go [hunk 3]
```diff
@@ -89,6 +125,98 @@ func NewValueOrRangeList[T ~[]uint16 | ~uint16 | ~string](v T) ValueOrRangeList
...
+		*p = append(*p, parsedPort)
+	}
+
+	return nil
+}
+
+func parsePort(s string) (Port, error) {
+	u, err := parseUint16(s)
+
+	if err != nil || u == 0 {
```
Comment (jijiechen) [inline pkg/transparentproxy/config/config.go:189]: Do we support specifying a `0` value to disable something?

Comment (bartsmykla) [reply pkg/transparentproxy/config/config.go:189]: No, I don't think we should as it would duplicate the existing logic and is unnecessary

**Review by jijiechen — APPROVED** [review]

**Comment by Icarus9913** [comment]
Could you please add a precedence explanation for the flags, ENV, config-files in the `install_transparent_proxy.go` command `Long` message?

**Review by Icarus9913 — APPROVED** [review]

**Comment by bartsmykla** [comment]
> Could you please add a precedence explanation for the flags, ENV, config-files in the `install_transparent_proxy.go` command `Long` message?

Unfortunately it's currently not trivial to do because of the library we are using to parse this stuff. We'll have to update our docs at the beginning. Maybe I'll have some time later to figure this out

File: pkg/transparentproxy/config/config.go (hunk 2 above)
Comment (bartsmykla) [reply pkg/transparentproxy/config/config.go]: That was to allow user to provide a value with an object with a `uid` key as basically it's how the underlying structure looks like, but I got rid of it

File: pkg/config/loader.go (hunk 1 above)
Comment (bartsmykla) [reply pkg/config/loader.go]: I actually changed this logic and got rid of `Filename` and `Content` properties. We are calling this function here: https://github.com/kumahq/kuma/blob/c15478c28567dc156cabbf7a6e60d1b345787ff1/app/kumactl/cmd/install/install_transparent_proxy.go#L140-L144

**Pull request merged**